### PR TITLE
gay: init at 1.2.8

### PR DESCRIPTION
**Pull request by l3gacyb3ta**
###### Motivation for this change

I love this package, and it wasn't in nixpkgs, so noone else could install it :(

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by l3gacyb3ta** [comment]
oops forgot to run formater

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1546,6 +1546,10 @@ with pkgs;
 
   fxlinuxprintutil = callPackage ../tools/misc/fxlinuxprintutil { };
 
+  gay = callPackage ../tools/misc/gay {
+    inherit (python38Packages) buildPythonPackage fetchPypi;
```
Comment (bobby285271) [inline pkgs/top-level/all-packages.nix:1550]: Will `python3Packages` be okay?

**Comment by bobby285271** [comment]
Will be great to add your maintainer information in a separate commit `maintainers: add l3gacyb3ta`, and follow https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md when writing commit messages.

File: pkgs/tools/misc/gay/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/ms-jpq/gay";
+    description = "Colour your text / terminal to be more gay.";
+    longDescription = ''
+      Colour your text / terminal to be more gay.
+      Gayer version of lolcat.
+    '';
+    license = licenses.publicDomain;
```
Comment (bobby285271) [inline pkgs/tools/misc/gay/default.nix:21]: [Maybe `licenses.mit`](https://github.com/ms-jpq/gay/blob/%3C3/LICENSE)? I am not sure.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134375` at 8bcad47e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>gay</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/tools/misc/gay/default.nix:16:0:
```
   |
16 |     description = "Colour your text / terminal to be more gay.";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/tools/misc/gay/default.nix (hunk 2 above)
Comment (yayayayaka) [reply pkgs/tools/misc/gay/default.nix:21]: Yup, change it to MIT.

**Review by yayayayaka — CHANGES_REQUESTED** [review]
Successfully tested execution of the binaries on nixos-21.05-x86_64.

Please rename the pull request title to `gay: init at 1.2.8`

Thanks for making the nixpkgs slightly more gay :3

File: pkgs/tools/misc/gay/default.nix (hunk 2 above)
Comment (l3gacyb3ta) [reply pkgs/tools/misc/gay/default.nix:21]: Done! Should be in next commit.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (l3gacyb3ta) [reply pkgs/top-level/all-packages.nix:1550]: Should be!

**Comment by l3gacyb3ta** [comment]
ok.... that might be it?

**Comment by l3gacyb3ta** [comment]
Still not the best at git sorry :)

**Comment by bobby285271** [comment]
You should rebase your work then force push, the commit log is too messy now.

Since this is not big diff, maybe `git rebase -i 57b6c8c5b280ac06e1005ba92592b5a213d96695`, drop all the commits listed, readd your package and maintainer info and commit, then `git push origin master -f`.

**Comment by l3gacyb3ta** [comment]
Ok I'll do this is a bit, thank!

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (davidak) [reply pkgs/top-level/all-packages.nix:1550]: i think that line is not needed. see how other packages do it (using pkgs)

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -1546,8 +1546,16 @@ with pkgs;
...
 
+  gay = callPackage ../tools/misc/gay {
+    inherit (python3Packages) buildPythonPackage fetchPypi;
+  };
+
   genann = callPackage ../development/libraries/genann { };
 
+  gay = callPackage ../tools/misc/gay {
+    inherit (python3Packages) buildPythonPackage fetchPypi;
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1557]: This is duplicated and also it is easier to use python3Packages.buildPythonPackage in the file itself.

File: pkgs/tools/misc/gay/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+buildPythonPackage rec {
+  pname = "gay";
+  version = "1.2.8";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "07ay0xjjjrm7qxz4c10s5mn2029rqq4m5ai26ypi5dh91al0g0pz";
+  };
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/gay/default.nix:11]: Can we run the tests inside the tests directory?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash the package commits together. There should be two commits in the end.

**Comment by l3gacyb3ta** [comment]
I'm not on nix atm, so I really cant test this without a lot of the work.

**Pull request closed without merging**